extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -0,0 +1,128 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import javax.inject.Inject;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
+import io.quarkus.deployment.index.ClassPathArtifactResolver;
+import io.quarkus.deployment.index.ResolvedArtifact;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.swaggerui.runtime.SwaggerUiTemplate;
+import io.quarkus.undertow.deployment.ServletExtensionBuildItem;
+
+public class SwaggerUiProcessor {
+
+    private static final Logger log = Logger.getLogger(SwaggerUiProcessor.class.getName());
+
+    private static final String SWAGGER_UI_WEBJAR_GROUP_ID = "org.webjars";
+    private static final String SWAGGER_UI_WEBJAR_ARTIFACT_ID = "swagger-ui";
+    private static final String SWAGGER_UI_WEBJAR_PREFIX = "META-INF/resources/webjars/swagger-ui";
+    private static final String SWAGGER_UI_DEFAULT_API_URL = "https://petstore.swagger.io/v2/swagger.json";
+    private static final String OPEN_API_DEFAULT_URL = "/openapi";
+    private static final String TEMP_DIR_PREFIX = "quarkus-swagger-ui_" + System.nanoTime();
+
+    /**
+     * The configuration for swagger-ui.
+     */
+    private SwaggerUiConfig swaggerUiConfig;
+
+    @Inject
+    private LaunchModeBuildItem launch;
+
+    /**
+     * Register this extension as a swagger-ui feature
+     *
+     * @return
+     */
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FeatureBuildItem.SWAGGER_UI);
+    }
+
+    /**
+     * Register the Swagger UI servlet extension
+     *
+     * @param template - Swagger UI runtime template
+     * @param container - the BeanContainer for creating CDI beans
+     * @return servlet extension build item
+     */
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    public void registerSwaggerUiServletExtension(SwaggerUiTemplate template,
+            BuildProducer<ServletExtensionBuildItem> servletExtension,
+            BeanContainerBuildItem container,
+            ShutdownContextBuildItem shutdown) {
+        if (launch.getLaunchMode().isDevOrTest()) {
+            try {
+                ResolvedArtifact artifact = getSwaggerUiArtifact();
+                Path tempDir = Files.createTempDirectory(TEMP_DIR_PREFIX);
+                extractSwaggerUi(artifact, tempDir);
```

Comment:
We really only need to do this once per JVM execution. I think this should be cached in a static, and use a Shutdown hook to clean it up. We don't really want to have to do this unzip after every code change.